SEP-31: include some details about the Sending and Receiving Anchors to make the SEP clearer (#962)
stellar/stellar-protocol: ecosystem/sep-0031.md

1. The Sending Anchor makes [SEP-12 PUT /customer](sep-0012.md#customer-put) requests containing the collected information for each client.
1. On successful registration (`202` HTTP status), the Sending Anchor makes a `POST /transactions` request to create a transaction record with the Receiving Anchor. This request contains the `id`s returned by the `PUT /customer` requests, the `transaction.fields` values collected from the Sending Client, as well as the transaction amount and asset information. The response will include a `id` that will be used to check the status of the transaction.
1. The Sending Anchor makes `GET /transactions/:id` requests until the transaction's `status` is `pending_sender`. 

> 1. The Sending Anchor submits the payment transaction to Stellar using the information provided in the `POST /transactions` response.

Reviewer: ❗ What information will it use? This is somewhat vague. If the intent is to say to send the an address that was specified in the post, it might be worth being more specific about that.

Reviewer 2: The response includes the destination account and the memo to use in the transaction. I felt we should refer to the endpoint that describes those attributes instead of explicitly mentioning them here, which we don't do for the other calls described in this section.

Author: IMO if we describe all fields here the text tends to become outdated as new fields are included so it's easier to maintain this spec if we direct the user to the endpoint containing the desired information.

Reviewer: Makes sense, thanks for explaining.

> 1. The Sending Anchor submits the payment transaction to Stellar using the memo and memo_type specified by the Receiving Anchor `GET /transactions/:id` response fields `stellar_memo` and `stellar_memo_type`.

Reviewer 2: Since this clarification leaves out the `stellar_account_id` response parameter, I think it's better to say the following:

```suggestion
1. The Sending Anchor submits the payment transaction to Stellar using the information provided in the `POST /transactions` response.
```

Author: https://github.com/stellar/stellar-protocol/pull/962/commits/37e9d9030c066f8aa6826509a17a68f0eda64fd6

Name | Type | Description
-----|------|------------

> `id` | string | The persistent identifier to check the status of this payment transaction.

Reviewer: ❔ What distinction are we making by specifying this as the `Stellar payment transaction` vs the `Stellar payment`? We refer to the payment as simply a Stellar payment elsewhere in the doc, like two lines down.

Author: For this SEP there's kind of an intersection between payment and transaction definitions even though they are distinct Stellar concepts.

The motivation for that change is the text said this is a "payment id" but it's returned from the `/transactions` endpoint, so it made sense from a syntax perspective to clarify that. That change may be unnecessary though.

`amount_in` | string | (optional) The amount received or to be received by the Receiving Anchor. Up to 7 decimals are permitted. Excludes any fees charged after Receiving Anchor receives the funds.
`amount_out` | string | (optional) The amount sent by the Rending Anchor to Receiving Client. Up to 7 decimals are permitted.
`amount_fee` | string | (optional) The amount of fee charged by the Receiving Anchor.

> `stellar_account_id` | string | The Receiving Anchor's Stellar account the Sending Anchor will be making the payment to.

Reviewer: 💡 I could be wrong, but when I read this my head automatically inserted this word between these two, then I realized it wasn't there. 
```suggestion
`stellar_account_id` | string | The Receiving Anchor's Stellar account that the Sending Anchor will be making the payment to.
```

`stellar_memo_type` | string | The type of memo to attach to the Stellar payment: `text`, `hash`, or `id`.
`stellar_memo` | string | The memo to attach to the Stellar payment.
`started_at` | UTC ISO 8601 string | (optional) Start date and time of transaction.
`completed_at` | UTC ISO 8601 string | (optional) Completion date and time of transaction.

> `stellar_transaction_id` | string | (optional) The transaction_id on Stellar network of the transfer that initiated the payment from the Sending Anchor to the Receiving Anchor.

Reviewer 2: I think this change is a bit redundant. The only Stellar transaction involved in a SEP-31 flow is the payment between these two parties, I don't see a reason to mention them here in addition to the `stellar_account_id` definition.

Author: I agree that's somewhat implicit but it wasn't very clear to me in my first read, that's why I added it. I think it's ok to leave it out of this update though.

Author: https://github.com/stellar/stellar-protocol/pull/962/commits/c7d4c260aa3c31b095a203a1fdc0138a5c271c53

`external_transaction_id` | string | (optional) The ID of transaction on external network that completes the payment into the receivers account.

> `refunded` | boolean | (optional) Should be true if the transaction was refunded. Refunded transactions should also have their `status` set to `error`. Not including this field means the transaction was not refunded.

Reviewer: ❔ Why do refund transactions have a status of error?

Reviewer 2: There isn't a better status for it. In retrospect we should have used a `refunded` status instead of a boolean here.

> `refunded` | boolean | (optional) Should be true if the transaction was refunded, and accompanied by the status `"error"` to make sure the Sending Anchor won't try to handle this transaction in any way. Not including this field means the transaction was not refunded.

Reviewer 2: ```suggestion
`refunded` | boolean | (optional) Should be true if the transaction was refunded. Refunded transactions should also have their `status` set to `error`. Not including this field means the transaction was not refunded.
```

Author: https://github.com/stellar/stellar-protocol/pull/962/commits/993c85e7f02c1bf8ddc336b142df5104690f2747
